### DISCUSSION TITLE
`useEventSource` no option to just watch specific events

### DISCUSSION THREAD

**Discussion by Shinigami92** [vueuse/vueuse#973] [General]
Currently you can register for some `events`, but still all events are triggered and passed to `data`.  
What is the best practice to handle specific events and observe the data for only such an event? So e.g. the used `data`-`watch` don't get triggered for every event.

**Comment by Shinigami92** [comment]
@wheatjs I think this issue can be converted into a discussion. I don't have the permission to do that 🙂

**Reply by wheatjs** [reply]
Done!

**Comment by atinux** [comment]
Agree with @Shinigami92 

I would like to refactor this part to use `useEventSource` but does not seem easy so far: https://github.com/Atinux/discuss/blob/147b6e58eddab29ab9ae94296a55e7cb71d6d9d2/components/Messages.vue#L13-L35

**Comment by atinux** [comment]
Would be nice also if there is a way to parse the data automatically here: https://github.com/vueuse/vueuse/blob/main/packages/core/useEventSource/index.ts#L57

Using [destr](https://github.com/unjs/destr) would be best
